File: Python/Evaluate Reverse Polish Notation.py
class Solution:
    # @param tokens, a list of string
    # @return an integer
    def evalRPN(self, tokens):
        seq = []
        for item in tokens:
            if item == "+":
                cnt1 = seq.pop()
                cnt2 = seq.pop()
                seq.append(cnt1 + cnt2)
            elif item == "-":
                cnt1 = seq.pop()
                cnt2 = seq.pop()
                seq.append(cnt2 - cnt1)
            elif item == "*":
                cnt1 = seq.pop()
                cnt2 = seq.pop()
                seq.append(cnt1 * cnt2)
            elif item == "/":
                cnt1 = seq.pop()
                cnt2 = seq.pop()
                if cnt1 * cnt2 < 0:
                    seq.append(-(abs(cnt2) / abs(cnt1)))
                else:
                    seq.append(cnt2 / cnt1)
            else:
                seq.append(int(item))
        return seq[0]

File: C++/Evaluate-Reverse-Polish-Notation.cpp
class Solution {
public:
    bool isop(const string &str) {
        if (str == "+" || str == "-" || str == "/" || str == "*") return true;
        return false;
    }
    int helper(int a, int b, const string &op) {
        if (op == "+") return a + b;
        else if (op == "*") return a * b;
        else if (op == "-") return b - a;
        else return b / a;
    }
    int evalRPN(vector<string>& tokens) {
        stack<int> stk;
        for (int i = 0; i < tokens.size(); i ++) {
            if (isop(tokens[i])) {
                int a = stk.top();
                stk.pop();
                int b = stk.top();
                stk.pop();
                stk.push(helper(a, b, tokens[i]));
            } else {
                stk.push(stoi(tokens[i]));
            }
        }
        return stk.top();
    }
};